packages/graphql-yoga/src/plugins/useSchema.ts

```diff
@@ -49,16 +49,26 @@ export const useSchema = <
   }
   const schemaByRequest = new WeakMap<Request, GraphQLSchema>()
   return {
-    async onRequest({ request }) {
+    async onParams({ request }) {
       const schema = await schemaDef(request)
       schemaByRequest.set(request, schema)
     },
     onEnveloped({ setSchema, context }) {
       if (context?.request) {
         const schema = schemaByRequest.get(context.request)
-        if (schema) {
-          setSchema(schema)
+        if (schema == null) {
+          throw new GraphQLError(
+            `No schema found for this request. Make sure you use this plugin with GraphQL Yoga.`,
```

Comment:
Should this message leak to the client?

Reply:
You're right. I changed it to Error now.

Reply:
Also maybe we should have tests for these?

Reply:
I'll add tomorrow 👍